casanovo/casanovo.py

```diff
@@ -7,116 +7,294 @@
+        logger.info("Sequencing and evaluating peptides from:")
+        for peak_file in annotated_peak_path:
+            logger.info("  %s", peak_file)
+
+        runner.evaluate(annotated_peak_path)
+
+    logger.info("DONE!")
+
+
+@main.command(cls=_SharedParams)
+@click.argument(
+    "train_peak_path",
     required=True,
-    help="The file path with peak files for predicting peptide sequences or "
-    "training Casanovo.",
-    multiple=True,
+    nargs=-1,
+    type=click.Path(exists=True, dir_okay=False),
 )
 @click.option(
-    "--peak_path_val",
-    help="The file path with peak files to be used as validation data during "
-    "training.",
+    "-p",
+    "--validation_peak_path",
+    help="""
+    An annotated MGF file for validation, like from MassIVE-KB. Use this
+    option multiple times to specify multiple files.
+    """,
+    required=True,
     multiple=True,
-)
-@click.option(
-    "--config",
-    help="The file name of the configuration file with custom options. If not "
-    "specified, a default configuration will be used.",
     type=click.Path(exists=True, dir_okay=False),
 )
+def train(
+    train_peak_path: Tuple[str],
+    validation_peak_path: Tuple[str],
+    model: Optional[str],
+    config: Optional[str],
+    output: Optional[str],
+    verbosity: str,
+) -> None:
+    """Train a Casanovo model on your own data.
+
+    TRAIN_PEAK_PATH must be one or more annoated MGF files, such as those
+    provided by MassIVE-KB, from which to train a new Casnovo model.
+    """
+    output, config = setup(model, config, output, verbosity, True)
+    with ModelRunner(config, model) as runner:
+        logger.info("Training a model from:")
+        for peak_file in train_peak_path:
+            logger.info("  %s", peak_file)
+
+        logger.info("Using the following validation files:")
+        for peak_file in validation_peak_path:
+            logger.info("  %s", peak_file)
+
+        runner.train(train_peak_path, validation_peak_path)
+
+    logger.info("DONE!")
+
+
+@main.command()
+def version() -> None:
+    """Get the Casanovo version information"""
+    versions = [
+        f"Casanovo: {__version__}",
+        f"Depthcharge: {depthcharge.__version__}",
+        f"Lightning: {lightning.__version__}",
+        f"PyTorch: {torch.__version__}",
+    ]
+    sys.stderr.write("\n".join(versions) + "\n")
+
+
+@main.command()
 @click.option(
+    "-o",
     "--output",
-    help="The base output file name to store logging (extension: .log) and "
-    "(optionally) prediction results (extension: .mztab).",
+    help="The output configuration file.",
+    default="casanovo.yaml",
     type=click.Path(dir_okay=False),
 )
-def main(
-    mode: str,
+def configure(output: str) -> None:
+    """Generate a Casanovo configuration file to customize.
+
+    The casanovo configuration file is in the YAML format.
+    """
+    Config.copy_default(output)
+    sys.stderr.write(f"Wrote {output}\n")
```

Comment:
**question:** Same question here, why not use the logger?

Reply:
Yikes! I don't remember doing that, but I should fix it 😛

Reply:
Yes, maybe we can indeed use the logging, and separate the setup into (i) initializing the logging, and (ii) initializing the model? Then we can consistently use (i) for all commands.